tests/test_schema.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+
+"""Tests for the jsonschema node validation."""
+
+import unittest
+
+import pybel.dsl
+from pybel.schema import is_valid_node
+from pybel.testing.utils import n
+
+
+VALID_ENTITY = {
+    "namespace": "",
+    "name": "",
+    "identifier": ""
+}
+
+VALID_VARIANT = {
+    "kind": "",
+    "concept": VALID_ENTITY,
+    "code": "",
+    "pos": 0
+}
+
+VALID_ABUNDANCE = {
+    "function": "",
+    "concept": VALID_ENTITY,
+    "xrefs": [VALID_ENTITY, VALID_ENTITY],
+    "variants": [VALID_VARIANT, VALID_VARIANT]
+}
+
+NAMESPACE, NAME, IDENTIFIER = n(), n(), n()
+
+
+def build_abundance(variant_dict):
+    """Return an abundance dict with the given variant."""
+    abundance = VALID_ABUNDANCE.copy()
+    abundance["variants"] = [variant_dict, variant_dict]
+    return abundance
+
+
+class TestSchema(unittest.TestCase):
+    """Tests for the jsonschema node validation."""
+
+    def test_pure_abundances(self):
+        """Test validating abundance nodes."""
+        abundance = pybel.dsl.Abundance(namespace=NAMESPACE, name=NAME, identifier=IDENTIFIER)
+        protein = pybel.dsl.Protein(namespace=NAMESPACE, name=NAME,
+                                    identifier=IDENTIFIER)
+        self.assertTrue(is_valid_node(abundance))
+        self.assertTrue(is_valid_node(protein))
+
+    def test_variant_abundances(self):
+        """Test validating abundance nodes with variants."""
+        gene_mod = pybel.dsl.GeneModification(NAME, namespace=NAMESPACE, identifier=IDENTIFIER)
+        protein_sub = pybel.dsl.ProteinSubstitution('Abc', 1, 'Def')
+        protein_mod = pybel.dsl.ProteinModification(NAME, code='Abc', position=1,
+                                                    namespace=NAMESPACE, identifier=IDENTIFIER)
+        fragment = pybel.dsl.Fragment(1, 2)
+
+        node = pybel.dsl.Gene(NAMESPACE, name=NAME, identifier=IDENTIFIER)
+
+        gmod = node.with_variants(gene_mod)
+        self.assertTrue(is_valid_node(gmod))
+
+        psub = node.with_variants(protein_sub)
+        self.assertTrue(is_valid_node(psub))
+
+        pmod = node.with_variants(protein_mod)
+        self.assertTrue(is_valid_node(pmod))
+
+        frag = node.with_variants(fragment)
+        self.assertTrue(is_valid_node(frag))
+
+    def test_fusions(self):
+        """Test validating fusion nodes."""
+        protein = pybel.dsl.Protein(namespace=NAMESPACE, name=NAME,
+                                    identifier=IDENTIFIER)
+        enum_fusion_range = pybel.dsl.EnumeratedFusionRange("r", 1, 79)
```

Comment:
does it validate that the fusion ranges are listed with the appropriate HGVS nomenclature? This might get a bit tricky, so let me know what you think.
Genes should have a "c" or "g", RNAs should have an "r", and proteins should have a "p"

Reply:
any idea on this one?

Reply:
Yep! I think I'll have to create separate schemas for each.

Reply:
Done!